vacasa/connect/connect.py

```diff
@@ -523,3 +523,36 @@ def get_reservations(self, params: dict = None):
         headers = self._headers()
 
         return self._iterate_pages(url, headers, params)
+
+    def get_quote_by_id(self, unit_id: int, arrival_date: str, departure_date: str, adult_count: int, params: dict = None):
+        """Retrieve a quote for a unit_id
+
+        Yields:
+            A dict containing attributes about the requested quote.
+        """
+
+        if params is None:
+            params = {}
+
+        if unit_id is None:
+            raise TypeError
```

Comment:
In the 4 spots where we're raising TypeError I think it would be good to add a message for the user that explains which argument caused the error.